alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/processor/service/ProcessorService.java

```diff
@@ -14,19 +14,28 @@
     protected final RetryingTransactionHelper retryingTransactionHelper;
     protected final ExecutorService executorService;
     protected final HealthProcessorConfiguration configuration;
+    protected final ProcessorAttributeService processorAttributeService;
 
     public void validateHealth() {
         if(!configuration.isEnabled()) {
             logger.info("Health validation initiated, but it is not enabled, aborting.");
             return;
         }
 
+        if(processorAttributeService
+                .getAttribute(ProcessorAttributeService.ATTR_KEY_IS_RUNNING, false)) {
+            logger.info("Health validation initiated, but it is already running, aborting.");
+            return;
+        }
+
         executorService.submit(() -> {
             logger.trace("Current thread id: {}", Thread.currentThread().getId());
             doInTransaction(
```

Comment:
Should the attribute not be persisted outside of the transaction?

Reply:
True, I assumed I need a write transaction to be able to set it :) 
Also, moved it outside of the thread scheduling to avoid concurrency issues.

Reply:
Scratch that, 5.x is safe, 6.x breaks.
Transaction **is** necessary.

> 2020-11-26 12:49:54,987  ERROR [quartz.core.JobRunShell] [DefaultScheduler_Worker-1] Job DEFAULT.healthProcessorTask threw an unhandled Exception: 
 org.springframework.scheduling.quartz.JobMethodInvocationFailedException: Invocation of method 'execute' on target class [class eu.xenit.alfresco.processor.tasks.ProcessorTask] failed; nested exception is java.lang.RuntimeException: Failed to create SAVEPOINT: createPropertyUniqueContext
	at org.springframework.scheduling.quartz.MethodInvokingJobDetailFactoryBean$MethodInvokingJob.executeInternal(MethodInvokingJobDetailFactoryBean.java:276)
	at org.springframework.scheduling.quartz.QuartzJobBean.execute(QuartzJobBean.java:75)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: java.lang.RuntimeException: Failed to create SAVEPOINT: createPropertyUniqueContext
	at org.alfresco.repo.domain.control.ibatis.ControlDAOImpl$PostgreSQL.createSavepoint(ControlDAOImpl.java:117)
	at org.alfresco.repo.domain.propval.AbstractPropertyValueDAOImpl.createPropertyUniqueContext(AbstractPropertyValueDAOImpl.java:1229)
	at org.alfresco.repo.attributes.AttributeServiceImpl.createAttribute(AttributeServiceImpl.java:211)
	at eu.xenit.alfresco.processor.service.ProcessorAttributeService.persistAttribute(ProcessorAttributeService.java:56)
	at eu.xenit.alfresco.processor.service.ProcessorService.validateHealth(ProcessorService.java:35)